### PR TITLE
Support Runtime/Platforms Downloading and Install 🚀 

### PR DESCRIPTION
**Pull request by MattKiazyk**
With Apple's change of splitting out the runtimes outside the of the main Xcode download, we need to support that in Xcodes so that users can still feel the speed and ease of downloading GB more of _required_ data

The Xcode CLI had support for this for a while.. sorry it took long to get this in. 

### XcodesKit
I started splitting out shared code into a new package called... **XcodesKit** so that, in the future, both CLI and UI can reuse the same code. Currently there's a lot of duplicated code. There are lots of shell calls that are exactly the same for both. 

Any new code that can be shared between the 2 should be moved to XcodesKit when at all possible

### Runtime Download

Xcodes CLI currently only uses the current selected Xcode to determine if it's installed. Anytime you have multiple Xcodes installed, it doesn't do a great job of showing you what _other_ runtimes are installed.

This PR includes code that uses `/Library/Developer/CoreSimulator/Images` (where the downloaded runtimes exist) to determine whether or not you have that it installed. 

I'm still working on tweaking the UI for downloading but currently looks like: 

| Installed | Downloading |
|--------|--------|
| Screenshot 2023-11-23 at 2 46 28 PM | Screenshot 2023-11-23 at 2 46 50 PM | 

This code started in June - so please give it a run if you have time.

### CODE REVIEW

File: Xcodes/Backend/AppState+Runtimes.swift [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+////                            self.error = error
+////                            self.presentedAlert = .generic(title: localizeString("Alert.Install.Error.Title"), message: error.legibleLocalizedDescription)
+////                        }
+////                        if let index = self.allXcodes.firstIndex(where: { $0.id == id }) {
+////                            self.allXcodes[index].installState = .notInstalled
+////                        }
+//                    }
+//                },
+//                receiveValue: { _ in }
+//            )
```
Comment (marcusziade) [inline Xcodes/Backend/AppState+Runtimes.swift:87]: Should we leave this in with a todo comment or remove it entirely?

File: Xcodes/Backend/Environment.swift [hunk 2]
```diff
@@ -112,6 +112,74 @@ public struct Shell {
         return (progress, publisher)
     }
     
+//    public var downloadWithAria2Async: (Path, URL, Path, [HTTPCookie]) async throws -> Progress = { aria2Path, url, destination, cookies in
```
Comment (marcusziade) [inline Xcodes/Backend/Environment.swift:115]: Should we add a TODO here or remove?

File: Xcodes/Frontend/InfoPane/RuntimesView.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                        .font(.headline)
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                    
+                    let builds = xcode.sdks?.allBuilds()
+                    let runtimes = builds?.flatMap { sdkBuild in
+                        appState.downloadableRuntimes.filter {
+                            $0.sdkBuildUpdate == sdkBuild
+                        }
+                    }
+        //            let runtimes = appState.getRunTimes(xcode: xcode)
```
Comment (marcusziade) [inline Xcodes/Frontend/InfoPane/RuntimesView.swift:27]: Is this needed?

**Review by marcusziade — APPROVED** [review]
lgtm

**Comment by hstdt** [comment]
Network change error makes `ObservingProgressIndicator` freezing until `InfoPane` reload.

File: Xcodes/Backend/Environment.swift (hunk 2 above)
Comment (MattKiazyk) [reply Xcodes/Backend/Environment.swift:115]: Added TODO to support Aria 2 with AsyncStream/Sequence

**Comment by MattKiazyk** [comment]
@hstdt thanks for the feedback. Going to work on supporting cancelling for runtimes in a future PR as I def missed that in this pr.

**Pull request merged**